### DISCUSSION TITLE
Embed within parent Click or Typer project

### DISCUSSION THREAD

**Discussion by rupurt** [tconbeer/harlequin#408] [Q&A, Troubleshooting, Support] [answered]
Howdy 👋 

Do you have any suggestions on how to embed `harlequin` within an existing Click or Typer application? I'm trying to use the `harlequin.cli` module put you can't pass any args to it and it doesn't pick up the `.harlequin.toml` config file.

Cheers

**Comment by tconbeer** [comment, marked as answer]
You can't use Click's decorator syntax to wrap a function that calls `harlequin.cli::build_cli`, since you need to execute the callable returned by `build_cli()`, not the function that wraps it. In other words, this will NOT work:

```py
import click

from harlequin.cli import build_cli


@click.group()
def my_group():
    pass

@my_group.command()
def hi():
    print("hi")

@my_group.command()
def harlequin():
    build_cli()()

if __name__ == "__main__":
    my_group()
```
As you noted, in this case `python my_group.py harlequin` will start Harlequin, but `python my_group.py harlequin --help` or other options won't work, because the `harlequin` command we defined above doesn't take any options.

Fortunately, Click has another API to register a command with a group, and that works. You pass in the callable returned by `build_cli()`, which is a click.Command:

```python
import click

from harlequin.cli import build_cli


@click.group()
def my_group():
    pass

@my_group.command()
def hi():
    print("hi")

my_group.add_command(build_cli(), "harlequin")

if __name__ == "__main__":
    my_group()
```

Now `python my_group.py harlequin my.db -t monokai` (etc) will work as intended.

**Reply by rupurt** [reply]
Awesome. I'll give it a try :crossed_fingers:
